### PR TITLE
#10/backend define types

### PR DESCRIPTION
**Pull request by TheHuns**
#10/backend define types
Set up the basic type definitions and updated server. All contained in one file right now but looking into how to modularize it in the future as it grows.

---
For now you can get server running and open at http://localhost:5000/graphql to easily inspect.
- See docs tab on the right to see possible queries and the one mutation  `addLocation`
- Added some example queries to README, the the playground page it also autocompletes for you if you want to try some
- For now queries should just return an empty array until we get some sample data ready for mongodb

### CODE REVIEW

File: api/server.js [hunk 1]
```diff
@@ -20,14 +16,88 @@ mongoose.connection.once("open", () => {
...
+    locationId: ID!
+    location: Location
+    state: String!
+    zip: Int
+    hours: Hours
+    parking: String
+  }
+
+  type Hours {
+    monday: [Int]
```
Comment (andreareed) [inline api/server.js:58]: I don’t see any keys on here. How will we connect them to a location?

File: api/server.js [hunk 2]
```diff
@@ -20,14 +16,88 @@ mongoose.connection.once("open", () => {
...
+
+  type Venue {
+    id: ID!
+    businessName: String!
+    category: String!
+    website: String!
+    phone: String!
+    address: String!
+    locationId: ID!
+    location: Location
```
Comment (andreareed) [inline api/server.js:48]: What’s the difference between address, location, and locationId?

File: api/models/location.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+const mongoose = require("mongoose");
+const Schema = mongoose.Schema;
+
+const locationSchema = new Schema({
```
Comment (andreareed) [inline api/models/location.js:4]: What is this being used for? What does `name` represent?

**Review by andreareed — COMMENTED** [review]
Nice! Just some questions, forgive my NoSQL ignorance 😂 

Also can you edit your PR comment and put the issue number in there?

File: api/models/location.js (hunk 3 above)
Comment (TheHuns) [reply api/models/location.js:4]: This is the mongodb schema setup(using mongoose pkg), figured name could be a city, county, geographical region, etc.

Comment (andreareed) [reply api/models/location.js:4]: This is 100% my own ignorance of Mongo, so sorry for hammering you with questions, just trying to understand more of what’s happening. How is this useful? What will be stored here and how will we access it?

File: api/server.js (hunk 1 above)
Comment (TheHuns) [reply api/server.js:58]: This is part of graphql i'm still trying to figure out and may change depending on the data structure our form sends. But i was trying to create an object "type" to be a field of another type. In the end should all be saved as one document to db

File: api/server.js (hunk 2 above)
Comment (TheHuns) [reply api/server.js:48]: address would be street address
locationId is the id of the location(city)
location would pull in the info about that city including other business' if its part of the query

File: api/models/location.js (hunk 3 above)
Comment (TheHuns) [reply api/models/location.js:4]: Your good, sorry I know I should comment more. This schema with mongoose sets up the fields for the documents we're trying to save to db and at the same time creates the class with methods we can use to do things like save, find, findByID, delete all real easy with this schema. some basic examples in the server.js

**Comment by TheHuns** [comment]
OK, had to go a little further than intended for this issue but made new commit where I believe the relationships between the data are set properly. Please give it a shot and let me know!  updated readme with more instructions

**Pull request merged**